shared/src/main/scala/kinesis4cats/producer/fs2/FS2Producer.scala

```diff
@@ -111,33 +112,36 @@ abstract class FS2Producer[F[_], PutReq, PutRes](implicit
         .evalMap { x =>
           val c = ctx.addEncoded("batchSize", x.size)
           x.toNel
-            .fold(F.unit) { records =>
+            .fold(F.unit) { x =>
+              val records = x.map(_._1)
+              val deferreds = x.map(_._2)
```

Comment:
**suggestion:** 
```suggestion
              val (records, deferreds) = x.separate
```

Reply:
I'm not sure I can because I don't think NonEmptyList has an Alternative instance

Reply:
ah, didn't check the types. Alright then